In [1]:
## imports vader
from vaderSentiment.vaderSentiment import SentimentIntensityAnalyzer


In [2]:
# imports library to perform statistical analysis like correlation
from scipy import stats

In [12]:
##Adding texts for sentiment analysis
group = ["Hii Kalpita",      
            "Sagar here",   
            "Hi Sagar",       
            "So what was it that you wanted to ask?",  
            "How did you approach for your RAs and TAs",  
            "So you'll probably get mails from your college if there are any open positions.If not then check your college's website where you can find job postings",
            "In my case, I emailed around 25 professors from my department with my CV.",
             "Oh that's something smart",
             "So out of those, one of them replied back and things went ahead",
             "Enough money for a month's expenses?",
             "Yup. It depends on where you live actually",
             "That was once you finalized right? Or after you were on campus finally",
             "In my case, my rent is 340 and I earn 540 per month so I manage",
             "Yeah because even if I get NEU, I'm not sure I'd join",
             "It's crazy competition",
             "Yeah I know right?",
             "Okay, I have to go now. I'll talk to you later. All the best",
             "Okay! Thanks Kalpita. Bye"]

In [4]:
## calls vader
analyzer = SentimentIntensityAnalyzer()

In [5]:
## analyzes sentiment for group 
positivesg = []
negativesg = []
for txt in group:
    vs = analyzer.polarity_scores(txt)
    print("{:-<65} {}".format(txt, str(vs)))
    positivesg.append(vs['pos'])
    negativesg.append(vs['neg'])

Hii Kalpita------------------------------------------------------ {'neg': 0.0, 'neu': 1.0, 'pos': 0.0, 'compound': 0.0}
Sagar here------------------------------------------------------- {'neg': 0.0, 'neu': 1.0, 'pos': 0.0, 'compound': 0.0}
Hi Sagar--------------------------------------------------------- {'neg': 0.0, 'neu': 1.0, 'pos': 0.0, 'compound': 0.0}
So what was it that you wanted to ask?--------------------------- {'neg': 0.0, 'neu': 1.0, 'pos': 0.0, 'compound': 0.0}
How did you approach for your RAs and TAs------------------------ {'neg': 0.0, 'neu': 1.0, 'pos': 0.0, 'compound': 0.0}
So you'll probably get mails from your college if there are any open positions.If not then check your college's website where you can find job postings {'neg': 0.0, 'neu': 1.0, 'pos': 0.0, 'compound': 0.0}
In my case, I emailed around 25 professors from my department with my CV. {'neg': 0.0, 'neu': 1.0, 'pos': 0.0, 'compound': 0.0}
Oh that's something smart---------------------------------------- 

In [7]:
### calculate Pearson's correlation
## prints correlation coefficient r, p-value
print(stats.pearsonr(positivesg,negativesg))

(-0.107029636061563, 0.6725093450185388)


### What sentiment analysis tool did you use (provide the name and a link). Describe how it identified sentiment in texts (e.g. classifier? rules? dictionaries?)

Ans: I use the VADER SENTIMENT analysis tool. Valence Aware Dictionary and Sentiment Reasoner(VADER) is a lexicon and rule based sentiment analysis tool that is specifically attenuated to sentiments expressed in social media. The lexicon based approach involves calculating orientation for a document from the semantic orientation (positive, negative or neutral) of words and phrases in the document. VADER has been found to be quite successful when dealing with social media texts, NY Times editorials, movie reviews, and product reviews. This is because VADER not only tells about the Positivity and Negativity score but also tells us about how positive or negative a sentiment is.
The tool can be found here: https://github.com/cjhutto/vaderSentiment#about-the-scoring 

### Tell us what type of online communication you analyzed.

Ans: I performed sentiment analysis on ‘Whatsapp messages’. The reason for choosing this form of CMC is because, it is easy to export messages and most often than usual, users use deception (particularly, butler lies) in casual conversations. I wanted to explore if this hypothesis was true using VADER. 

### Provide the data and the positivity and negativity scores for each message. Looking at the results at a text level, do you agree with every classification it made?

Ans: The compound score is a metric that is sued to denote whether a text is positive, negative or neutral. It does so by summing the valence scores of each word in the lexicon, adjusted according to the rules, and then normalized to be between -1 (most extreme negative) and +1 (most extreme positive). Typical threshold values (used in the literature cited on this page) are:
1.	positive sentiment: compound score >= 0.05
2.	neutral sentiment: (compound score > -0.05) and (compound score < 0.05)
3.	negative sentiment: compound score <= -0.05

#### RESULTS: 

1. Hii Kalpita--------- {'neg': 0.0, 'neu': 1.0, 'pos': 0.0, 'compound': 0.0} 
##### Compound score: 0 
##### Neutral score: 1 
This denotes that the sentiment is neither positive or negative. I agree with this classification.

.................

2. Sagar here---------- {'neg': 0.0, 'neu': 1.0, 'pos': 0.0, 'compound': 0.0}
##### Compound score: 0 
##### Neutral score: 1 
This denotes that the sentiment is neither positive or negative. I agree with this classification.

.................

3. Hi Sagar------------ {'neg': 0.0, 'neu': 1.0, 'pos': 0.0, 'compound': 0.0} 
##### Compound score: 0 
##### Neutral score: 1 
This denotes that the sentiment is neither positive or negative. I agree with this classification.

.................

4. So what was it that you wanted to ask?-----{'neg': 0.0, 'neu': 1.0, 'pos': 0.0, 'compound': 0.0} 
##### Compound score: 0 
##### Neutral score: 1 
This denotes that the sentiment is neither positive or negative. I agree with this classification.

.................

5. How did you approach for your RAs and TAs--------- {'neg': 0.0, 'neu': 1.0, 'pos': 0.0, 'compound': 0.0} 
##### Compound score: 0 
##### Neutral score: 1 
This denotes that the sentiment is neither positive or negative. I agree with this classification.

.................

6. So you'll probably get mails from your college if there are any open positions.If not then check your college's website where you can find job postings------------ {'neg': 0.0, 'neu': 1.0, 'pos': 0.0, 'compound': 0.0} 
##### Compound score: 0 
##### Neutral score: 1 
This denotes that the sentiment is neither positive or negative. I agree with this classification.

.................

7. In my case, I emailed around 25 professors from my department with my CV. ----------------------{'neg': 0.0, 'neu': 1.0, 'pos': 0.0, 'compound': 0.0} 
##### Compound score: 0 
##### Neutral score: 1 
This denotes that the sentiment is neither positive or negative. I agree with this classification.

.................

8. Oh that's something smart------- {'neg': 0.0, 'neu': 0.526, 'pos': 0.474, 'compound': 0.4019} 
##### Compound score: 0 .4019
##### Positive score: 0.474
Since the compound score > 0.05, the sentiment analysis for this text is positive. I agree with this classification.

.................

9. So out of those, one of them replied back and things went ahead-- {'neg': 0.0, 'neu': 1.0, 'pos': 0.0, 'compound': 0.0} 
##### Compound score: 0 
##### Neutral score: 1 
This denotes that the sentiment is neither positive or negative. I agree with this classification.

.................

10. Enough money for a month's expenses?----------------------------- {'neg': 0.0, 'neu': 1.0, 'pos': 0.0, 'compound': 0.0} 
##### Compound score: 0 
##### Neutral score: 1 
This denotes that the sentiment is neither positive or negative. I agree with this classification.

.................

11. Yup. It depends on where you live actually----------------------- {'neg': 0.0, 'neu': 1.0, 'pos': 0.0, 'compound': 0.0} 
##### Compound score: 0 
##### Neutral score: 1 
This denotes that the sentiment is neither positive or negative. I agree with this classification.

.................

12. That was once you finalized right? Or after you were on campus finally {'neg': 0.0, 'neu': 1.0, 'pos': 0.0, 'compound': 0.0} 
##### Compound score: 0 
##### Neutral score: 1 
This denotes that the sentiment is neither positive or negative. I agree with this classification.

.................

13. In my case, my rent is 340 and I earn 540 per month so I manage-- {'neg': 0.0, 'neu': 1.0, 'pos': 0.0, 'compound': 0.0} 
##### Compound score: 0 
##### Neutral score: 1 
This denotes that the sentiment is neither positive or negative. I agree with this classification.

.................

14. Yeah because even if I get NEU, I'm not sure I'd join------------ {'neg': 0.274, 'neu': 0.569, 'pos': 0.157, 'compound': -0.1655} 
##### Compound score: -0.1655
##### Negative score: 0.274
##### Positive score: 0.157
Since the compound score < -0.05, the sentiment analysis for this text is negative. I agree with this classification because the user feels negatively about this decision and the lexicon has accurately computed 'not sure' as a negative sentiment.

.................

15. It's crazy competition------------------------------------------- {'neg': 0.545, 'neu': 0.455, 'pos': 0.0, 'compound': -0.34} 
##### Compound score: - 0.34
##### Negative score: 0.545
Since the compound score < -0.05, the sentiment analysis for this text is negative. I disagree with this classification. Because the analysis might have computed 'crazy' as a negative emotion whereas it has just been used in this text to denote a magnitude or extent of competition. 

.................

16. Yeah I know right?----------------------------------------------- {'neg': 0.0, 'neu': 0.476, 'pos': 0.524, 'compound': 0.296} 
##### Compound score: 0 .296
##### Positive score: 0.524
Since the compound score > 0.05, the sentiment analysis for this text is positive. I agree with this classification because this text denotes excitement and the tool has accurately computed the sentiment as positive by scanning words like 'yeah','right'

.................

17. Okay, I have to go now. I'll talk to you later. All the best----- {'neg': 0.0, 'neu': 0.643, 'pos': 0.357, 'compound': 0.7269} 
##### Compound score: 0 7263
##### Positive score: 0.357
Since the compound score > 0.05, the sentiment analysis for this text is positive. I disagree with this classification because the user says 'I have to go, talk to you later'. Although these phrases may not seem negative, I feel that they denote lack of interest in continuing the conversation. The text ends with 'All the best' which is a positive sentiment and I strongy believe that the tool must have computed the score based on the word 'best' 

.................

18. Okay! Thanks Kalpita. Bye---------------------------------------- {'neg': 0.0, 'neu': 0.282, 'pos': 0.718, 'compound': 0.6239}
##### Compound score: 0 .6239
##### Positive score: 0.718
Since the compound score > 0.05, the sentiment analysis for this text is positive. I agree with this classification because the user is grateful. The use of the word 'Thanks' has been accurately computed as positive by the tool. 



### Do your data support the hypothesis that positivity and negativity are opposites. Why or why not? Why do you think you observe the pattern that you do? (Advanced - calculate a correlation and/or a test of significance to explain the relationship.)
Ans: No. My data does not support the hypothesis that positivity and negativity are opposites. Although there were certain cases in the results where the tool computed the analysis to be negative because of the use of certain words like 'crazy'. In context with this particuar conversation, the word 'crazy' was used to signify the magnitude of something. However, the lexcion value computed the sentiment as negative. Another case was when the tool computed the analysis as positive instead of negative. Phrases like 'I have to go', 'I'll talk to you later' more often than not denote disinterest in continuing the conversation. With respect to this conversation, the user didn't respond to the previous text that was a rhetorical question and decided to leave the conversation all of a sudden. This denotes lack of interest which wasn't computed as negative by the tool. Overall, the sentiment analysis was accurate for most of the texts. 

### Correlation 
The statistical result was as follows:
###### prints correlation coefficient r, p-value
###### (-0.107029636061563, 0.6725093450185388)
The correlation coefficient is -0.107 , which denotes strong negative association between positivity and negativity.

Null hypothesis: There is no difference between positivity and negativity
Alternative hypothesis: There is a difference between positivity and negativity

Since the pvalue(0.672) is greater than 0.05, we fail to reject the null hypothesis. Thus, there is not enough evidence to support the alternative hypothesis that positvity and negativity are opposites. 